### PR TITLE
Jac/no parsers from commands

### LINKED ISSUE #23: Create Group Command
Docs for Tabcmd - https://help.tableau.com/current/server/en-us/tabcmd_cmd.htm#idbfa546df-2be7-444f-8bb8-07978b127783

Tabcmd2 "create group" command contains flag --name as group name. To create a group Tabcmd2 uses `tabcmd2 creategroup --name "ExampleGroup"`. Original Tabcmd does not use --name flag. Update the Tabcmd2 command code accordingly.

### PR DESCRIPTION
**Pull request by jacalata**
Major changes
1 - restructure into a single main file (tabcmd.py) and all other files in a module, including a new module called 'execution' (not a great name, open to other ideas!)
- (Making absolute references to modules means we can stop listing things in the init files)

2 - edit all parsers to return values only by adding them as attributes on args
3 - add tests that call each Command

Next: replace our manual identification of which Command to call with letting argparse identify the right one (this will also fix #23 and all the other commands that need to have a name with no --option designation)

### CODE REVIEW

File: tabcmd/commands/datasources_and_workbooks/export_command.py [hunk 1]
```diff
@@ -1,129 +1,113 @@
...
+                    logger.info("Exported successfully")
 
             except TSC.ServerResponseError as e:
-                self.logger.error("Server error occurred")
+                ExportCommand.exit_with_error(logger, "Server Error:", e)
 
-        if self.args.pdf or self.args.png or self.args.csv:  # its a workbook
-            if self.args.pdf:  # its a view
-                view = self.get_view(self.url)
+        elif args.pdf or args.png or args.csv:  # its a workbook ????
```
Comment (ovinis) [inline tabcmd/commands/datasources_and_workbooks/export_command.py:59]: We can remove that comment. csv, pdf, png applies to view only.

File: tabcmd/commands/extracts/create_extracts_command.py [hunk 2]
```diff
@@ -11,52 +11,35 @@ class CreateExtracts(ExtractsCommand):
...
+                ExtractsCommand.exit_with_error(logger, "Server Error:", e)
+        elif args.workbook:
             try:
-                project_id = ProjectCommand.find_project_id(server,
-                                                            self.args.project)
-                workbook_item = ExtractsCommand.get_workbook_item(server,
-                                                                  self.args
-                                                                  .workbook)
+                # this isn't used anywhere?
+                project_id = ProjectCommand.find_project_id(server, args.project)
```
Comment (ovinis) [inline tabcmd/commands/extracts/create_extracts_command.py:35]: True, It is not used anywhere. We don't need to assign it to a variable.

File: tabcmd/parsers/login_parser.py [hunk 3]
```diff
@@ -22,20 +20,19 @@ def login_parser():
...
+            else:
+                logger.error("Please provide password")
+                sys.exit()
+        if args.token_name and not args.token:
+            if args.prompt:
+                args.token = getpass.getpass("Token:")
+            else:
+                logger.error("Please provide password")
+                sys.exit()
+
```
Comment (ovinis) [inline tabcmd/parsers/login_parser.py:35]: This block above is so much cleaner now. Thank you!

File: tests/commands/test_geturl_utils.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        assert filetype == 'pdf', filetype
+
+    def test_evaluate_file_name_url(self):
+        mock_logger = mock.MagicMock()
+        filename = None
+        url = 'long-url-stuff-goes-here.pdf'
+        filetype = GetUrl.evaluate_file_name(mock_logger, filename, url)
+        assert filetype == 'pdf', filetype
+
+    """
```
Comment (ovinis) [inline tests/commands/test_geturl_utils.py:22]: Are these the remaining tests we should write?

File: tests/e2e/credentials.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+SERVER_URL = "http://qa-near.tsi.lan/"
+SITE_NAME = "jac"
+PAT_NAME = "jac2"
+PAT = "F3ZysrkhR8Ovzm5YUoO7VA==:o41SzK9InZsNgx6vMvZGDUaoknw6IpZN"
```
Comment (ovinis) [inline tests/e2e/credentials.py:4]: Im guessing these are fake credentials for testing purposes?

File: tests/e2e/vars.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+"""
+This file is where usernames, project names, server address, etc can be defined
+These tests should be runnable live on Tableau Online or Tableau Server
+Assuming, of course, that you have a site or system administrator account.
+"""
+
+"""
+DO NOT #### add your credentials to this file.
+You can add them to credentials.py, which will never be checked in.
```
Comment (ovinis) [inline tests/e2e/vars.py:9]: I may have missed it, if so I apologize. I cant seem to find [tests/e2e/credentials.py] file under the gitignore file.

File: tests/commands/test_geturl_utils.py (hunk 4 above)
Comment (jacalata) [reply tests/commands/test_geturl_utils.py:22]: it's a list of all the methods in the file, we should have a unit test that at least executes each of them

File: tabcmd.spec [hunk 7]
```diff
@@ -0,0 +1,44 @@
+# -*- mode: python ; coding: utf-8 -*-
+
+
```
Comment (ovinis) [inline tabcmd.spec:3]: Is this spec related to building the python exe?

**Comment by ovinis** [comment]
How does the code handle cases where uses don't have to provide a flag. For example, in the Creategroup command of Tabcmd1, users can just type `tabcmd creategroup "Development"` instead of `tabcmd2 creategroup --name "ExampleGroup"`

**Review by ovinis — APPROVED** [review]

File: tabcmd.spec (hunk 7 above)
Comment (jacalata) [reply tabcmd.spec:3]: yes - currently it's all just the default settings from pyinstaller.
